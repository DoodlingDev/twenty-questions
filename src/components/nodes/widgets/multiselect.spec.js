/* eslint require-jsdoc: "off" */
import React from "react";
import MultiSelect from "./multiselect";
import renderer from "react-test-renderer";
import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

function setup(renderFn, testProps = defaultProps) {
  return renderFn(<MultiSelect {...testProps} />);
}

const defaultProps = {
  name: "test_multi",
  options: ["one", "two", "three"],
  path: "path.to.testing_component",
  valueManager: {
    values: {},
    validate: {},
    update: jest.fn(),
  },
};

describe("rendering", () => {
  it("renders correctly", () => {
    const render = renderer.create(<MultiSelect {...defaultProps} />).toJSON();
    expect(render).toMatchSnapshot();
  });

  it("renders correctly with an item checked", () => {
    const oneCheckedProps = {
      ...defaultProps,
      valueManager: {
        values: { "path.to.testing_component": ["two"] },
        validate: {},
        update: jest.fn(),
      },
    };
    const render = renderer.create(<MultiSelect {...oneCheckedProps} />).toJSON();
    expect(render).toMatchSnapshot();
  });
});

describe("updating", () => {
  it("removes item from state if checked -> unchecked", () => {
    const twoCheckedProps = {
      ...defaultProps,
      valueManager: {
        values: { "path.to.testing_component": ["one", "two"] },
        validate: {},
        update: jest.fn(),
      },
    };
    const wrapper = setup(mount, twoCheckedProps);
    wrapper
      .find("input")
      .first()
      .simulate("change", {
        currentTarget: {
          name: "one",
        },
      });
    expect(wrapper.props().valueManager.update.mock.calls[0][0].value).toEqual([
      "two",
    ]);
    wrapper.props().valueManager.update.mockReset();
  });

  it("removes item from state if unchecked -> checked", () => {
    const twoCheckedProps = {
      ...defaultProps,
      valueManager: {
        values: { "path.to.testing_component": ["two", "three"] },
        validate: {},
        update: jest.fn(),
      },
    };
    const wrapper = setup(mount, twoCheckedProps);
    wrapper
      .find("input")
      .first()
      .simulate("change");
    expect(wrapper.props().valueManager.update.mock.calls[0][0].value).toEqual([
      "one",
      "two",
      "three",
    ]);
    wrapper.props().valueManager.update.mockReset();
  });
});
