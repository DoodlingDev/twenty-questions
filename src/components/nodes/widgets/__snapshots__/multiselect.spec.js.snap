// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly 1`] = `
Array [
  <label
    htmlFor="path.to.testing_component-option-0"
  >
    <input
      checked={false}
      id="path.to.testing_component-option-0"
      name="one"
      onChange={[Function]}
      type="checkbox"
    />
    one
  </label>,
  <label
    htmlFor="path.to.testing_component-option-1"
  >
    <input
      checked={false}
      id="path.to.testing_component-option-1"
      name="two"
      onChange={[Function]}
      type="checkbox"
    />
    two
  </label>,
  <label
    htmlFor="path.to.testing_component-option-2"
  >
    <input
      checked={false}
      id="path.to.testing_component-option-2"
      name="three"
      onChange={[Function]}
      type="checkbox"
    />
    three
  </label>,
]
`;

exports[`rendering renders correctly with an item checked 1`] = `
Array [
  <label
    htmlFor="path.to.testing_component-option-0"
  >
    <input
      checked={false}
      id="path.to.testing_component-option-0"
      name="one"
      onChange={[Function]}
      type="checkbox"
    />
    one
  </label>,
  <label
    htmlFor="path.to.testing_component-option-1"
  >
    <input
      checked={true}
      id="path.to.testing_component-option-1"
      name="two"
      onChange={[Function]}
      type="checkbox"
    />
    two
  </label>,
  <label
    htmlFor="path.to.testing_component-option-2"
  >
    <input
      checked={false}
      id="path.to.testing_component-option-2"
      name="three"
      onChange={[Function]}
      type="checkbox"
    />
    three
  </label>,
]
`;
